volatility/framework/plugins/windows/netscan.py

```diff
@@ -0,0 +1,306 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import datetime
+from typing import Iterable, List
+
+from volatility.framework import constants, exceptions, interfaces, renderers, symbols, layers
+from volatility.framework.automagic.pdbscan import scan
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.symbols import intermed
+from volatility.framework.symbols.windows.extensions import network
+from volatility.plugins import timeliner
+from volatility.plugins.windows import poolscanner
+
+vollog = logging.getLogger(__name__)
+
+class NetScan(interfaces.plugins.PluginInterface):
+    """Scans for network objects present in a particular windows memory image."""
+
+    _version = (1, 0, 0)
+    CORRUPT_DEFAULT = False
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.PluginRequirement(name = 'poolscanner', plugin = poolscanner.PoolScanner, version = (1, 0, 0)),
+            requirements.BooleanRequirement(name = 'include-corrupt',
+                description = "Radically eases result validation. This will show partially overwritten data. WARNING: the results are likely to include garbage and/or corrupt data. Be cautious!",
+                default = cls.CORRUPT_DEFAULT,
```

Comment:
This should probably always be False, since otherwise there would likely be no way to turn if off?  (If it defaults to true and providing the flag to true, it's sort of fixed on).

Reply:
default is False (view `cls.CORRUPT_DEFAULT`) but it's set to optional=True - I took this as meaning that this flag is optional.

Reply:
It is, I just meant that defining CORRUPT_DEFAULT is a little bit redundant, since setting it to true would essentially force the flag on.

Reply:
Oh, you meant removing the class variable altogether? Makes sense, I'll do that.